src/bdc/steps/social_media_api.py

```diff
@@ -0,0 +1,72 @@
+# SPDX-License-Identifier: MIT
+# SPDX-FileCopyrightText: 2023 Ahmed Sheta <ahmed.sheta@fau.de>
+
+from http import HTTPStatus
+
+import facebook
+import pandas as pd
+import requests
+from tqdm import tqdm
+
+from bdc.steps.step import Step
+from config import FACEBOOK_APP_SECRET, FACEBOOK_SDK_GRAPH_API_KEY
+
+
+class FacebookGraphAPI(Step):
+    name = "Facebook_Graph"
+
+    def load_data(self) -> None:
+        pass
+
+    def verify(self) -> bool:
+        return (
+            self.df is not None and "First Name" in self.df and "Last Name" in self.df
+        )
+
+    def run(self):
+        self.df["Full Name"] = self.df["First Name"] + " " + self.df["Last Name"]
+        tqdm.pandas(desc="Searching Facebook")
+        try:
+            self.df["Full Name"].progress_apply(
+                lambda lead: self.search_facebook_graph(lead)
+            )
+        except facebook.GraphAPIError as e:

```

Comment:
This exception already being caught on line with another try catch block. Other types of errors (like pandas-related errors) could also occur here and would not be caught appropriately.

Reply:
Addressed.